cirq-core/cirq/linalg/transformations.py

```diff
@@ -29,7 +29,7 @@
 # user provides a different np.array([]) value.
 RaiseValueErrorIfNotProvided: np.ndarray = np.array([])
 
-_NPY_MAXDIMS = 32  # Should be changed once numpy/numpy#5744 is resolved.
+_NPY_MAXDIMS = 64
```

Comment:
Will this break if someone tries to use cirq with 1.x? Can we check the numpy major version at runtime to get the value instead of hard-coding it?

Reply:
updated to get the value from numpy `np.MAXDIMS`

Reply:
so numpy is not consitent about whether np.MAXDIMS is exposed or not accros versions

Reply:
Can you just derive it from `numpy.__version__`?

Reply:
We will run out of memory with non-trivial arrays way before getting to 64 dimensions.  I like option 1 or some variation of it.

Reply:
@maffoo I like option 2, but I will use np.empty instead

---

@pavoljuhas why?

```py3
>>> np.empty((1,)*64)
array([[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[0.0078125]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]) 
```

Reply:
> @pavoljuhas why?

`2 ** 64 ~= 1.8e19`

Reply:
that's if we create the a state vector ... but we are just testing whether the library supports a given number of dimensions ... this now can be up to 33 (~128GB)  but may reach 35 in a couple of years ... obviously talking about  anything higher than high 30s doesn't make sense